src/main/java/com/deloitte/elrr/services/controller/ClientTokenController.java

```diff
@@ -37,9 +57,67 @@ public ResponseEntity<ClientTokenDto> createToken(
             @Valid @RequestBody PermissionsWrapperDto wrapper)
             throws ResourceNotFoundException {
 
-        ClientTokenDto clientToken = new ClientTokenDto();
-        clientToken.setToken(jwtUtil.createToken(wrapper.getPermissions()));
+        UUID tokenId = Generators.timeBasedEpochRandomGenerator().generate();
+        String token = jwtUtil.createToken(tokenId, wrapper.getPermissions());
+        // get the payload back out
+        // TODO figure out a way to avoid this
+        DecodedJWT decodedJWT = jwtUtil.decodeToken(token);
+        Map<String, Object> payload = decodedJWT.getClaims().entrySet().stream()
+                .collect(Collectors.toMap(Map.Entry::getKey,
+                        e -> e.getValue().as(Object.class)));
+        ClientToken clientToken = new ClientToken();
+        // set the label if provided
+        if (wrapper.getLabel() != null && !wrapper.getLabel().isEmpty()) {
+            clientToken.setLabel(wrapper.getLabel());
+        }
+        clientToken.setJwtPayload(payload);
+        clientToken.setId(tokenId);
+        clientTokenSvc.save(clientToken);
+        // Output the token to the client
+        ClientTokenDto clientTokenDto = new ClientTokenDto();
+        clientTokenDto.setToken(token);
+
+        return ResponseEntity.ok(clientTokenDto);
+    }
+
+    /**
+     * Revoke (delete) a client token by its ID.
+     *
+     * @param tokenId The UUID of the token to revoke
+     * @return ResponseEntity with no content if successful
+     * @throws ResourceNotFoundException if token is not found
+     */
+    @DeleteMapping("/token/{tokenId}")
+    public ResponseEntity<Void> revokeToken(@PathVariable UUID tokenId)
+            throws ResourceNotFoundException {
+
+        try {
+            clientTokenSvc.delete(tokenId);
+            return ResponseEntity.noContent().build();
+        } catch (Exception e) {
+            throw new ResourceNotFoundException(
+                    "Token not found with ID: " + tokenId);
+        }
+    }
+
+    /**
+     * List all client tokens.
+     *
+     * @return ResponseEntity<List<ClientTokenListDto>> containing all
+     * tokens with their ID and label
+     */
+    @GetMapping("/tokens")
+    public ResponseEntity<List<ClientTokenListItemDto>> listTokens() {
+        Iterable<ClientToken> tokens = clientTokenSvc.findAll();
+        List<ClientTokenListItemDto> tokenList = new ArrayList<>();
+
+        for (ClientToken token : tokens) {
+            ClientTokenListItemDto dto = new ClientTokenListItemDto();
+            dto.setId(token.getId());
```

Comment:
we have something called modelMapper used in the other controllers, which does a name/type based mapping automatically between entities and DTOs in one line. This is a very small entity but worth a consideration.